e2/electron.vite.config.ts

```diff
@@ -1,15 +1,34 @@
-import { defineConfig, externalizeDepsPlugin } from 'electron-vite'
+import { defineConfig, externalizeDepsPlugin } from 'electron-vite';
 import { nodePolyfills } from "vite-plugin-node-polyfills";
-import { svelte } from '@sveltejs/vite-plugin-svelte'
+import { svelte } from '@sveltejs/vite-plugin-svelte';
+import esmShim from 'esm-shim-plugin';
 
 export default defineConfig({
   main: {
     plugins: [
       externalizeDepsPlugin({ exclude: ["@radically-straightforward/sqlite"] }),
-    ]
+      esmShim(),
+    ],
+    build: {
+      rollupOptions: {
+        output: {
+          format: 'es',
+        },
+      },
+    },
   },
   preload: {
-    plugins: [externalizeDepsPlugin()]
+    plugins: [
+      externalizeDepsPlugin(),
+      esmShim(),
+    ],
+    build: {
+      rollupOptions: {
+        output: {
+          format: 'es',
```

Comment:
Another alternative option would be to not use ESM for preload scripts, but to keep those as CJS.
Only the main would be ESM.
Only a suggestion, in case that makes things a lot easier.